### PR TITLE
UCT/IB/MD: Add new ways to detect GPUdirect support

### PR DESCRIPTION
**Pull request by ivankochin**
## What
Currently GPUdirect support detected by checking `/sys/kernel/mm/memory_peers/nv_mem/version`. This patch adds two new files that can be used to detect GPUdirect support: `/sys/module/nvidia_peermem/version` for driver version >= 470 and `/sys/module/nv_peer_mem/version` for driver version < 470.

## Why ?
The upstream linux kernel has dropped the `/sys/kernel/mm` paths since kernel 5.6. So we need to implement new ways to detect peermem support.

### CODE REVIEW

**Review by tvegas1 — APPROVED** [review]

**Review by brminich — APPROVED** [review]

File: src/uct/ib/base/ib_md.c [hunk 1]
```diff
@@ -1156,13 +1156,17 @@ void uct_ib_md_parse_relaxed_order(uct_ib_md_t *md,
...
+    }
+
     if (!access(file, F_OK)) {
         md->reg_mem_types |= UCS_BIT(mem_type);
     }
 
-    ucs_debug("%s: %s GPUDirect RDMA is %s", uct_ib_device_name(&md->dev),
-              ucs_memory_type_names[mem_type],
-              md->reg_mem_types & UCS_BIT(mem_type) ? "enabled" : "disabled");
+    ucs_debug("%s: %s GPUDirect RDMA is %sdetected by checking %s file",
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.c:1167]: "... by checking %s",

File: src/uct/ib/base/ib_md.c [hunk 2]
```diff
@@ -1235,10 +1239,18 @@ ucs_status_t uct_ib_md_open_common(uct_ib_md_t *md,
     /* Check for GPU-direct support */
     md->reg_mem_types = UCS_BIT(UCS_MEMORY_TYPE_HOST);
     if (md_config->enable_gpudirect_rdma != UCS_NO) {
-        /* check if GDR driver is loaded */
+        /* Check if GDR driver is loaded. Different driver/kernel configurations
+           can have different ways to check it */
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.c:1243]: /* Check peer memory driver is loaded, different driver versions use different paths */

Comment (ivankochin) [reply src/uct/ib/base/ib_md.c:1243]:
```suggestion
        /* Check peer memory driver is loaded, different driver versions use different paths */
```

File: src/uct/ib/base/ib_md.c (hunk 1 above)
Comment (ivankochin) [reply src/uct/ib/base/ib_md.c:1167]:
```suggestion
    ucs_debug("%s: %s GPUDirect RDMA is %sdetected by checking %s",
```

File: src/uct/ib/base/ib_md.c [hunk 3]
```diff
@@ -1239,8 +1239,7 @@ ucs_status_t uct_ib_md_open_common(uct_ib_md_t *md,
     /* Check for GPU-direct support */
     md->reg_mem_types = UCS_BIT(UCS_MEMORY_TYPE_HOST);
     if (md_config->enable_gpudirect_rdma != UCS_NO) {
-        /* Check if GDR driver is loaded. Different driver/kernel configurations
-           can have different ways to check it */
+        /* Check peer memory driver is loaded, different driver versions use different paths */
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.c:1242]: pls split long line and squash

Comment (ivankochin) [reply src/uct/ib/base/ib_md.c]: Done

**Review by brminich — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**